Sources/Sentry/SentryProfiler.mm

```diff
@@ -312,12 +312,9 @@ + (void)startWithHub:(SentryHub *)hub
 
     [_gCurrentProfiler start];
 
-    _gCurrentProfiler->_timeoutTimer = [SentryDependencyContainer.sharedInstance.timerFactory
-        scheduledTimerWithTimeInterval:kSentryProfilerTimeoutInterval
-                                target:self
-                              selector:@selector(timeoutAbort)
-                              userInfo:nil
-                               repeats:NO];
+    // from NSTimer.h: Timers scheduled in an async context may never fire.
+    dispatch_async(dispatch_get_main_queue(), ^{ [self scheduleTimeoutTimer]; });
```

Comment:
Looks like it, yeah! See the updated implementation where I check `+[NSThread isMainThread]`, because if you dispatch_async to the main queue when you're already on it, and the caller blocks the main queue, it'll never even get scheduled!

Reply:
Done in https://github.com/getsentry/sentry-cocoa/pull/3138